app/scripts/migrations/068.test.js

```diff
@@ -0,0 +1,2491 @@
+import migration68 from './068';
+
+describe('migration #68', () => {
+  it('should update the version metadata', async () => {
+    const oldStorage = {
+      meta: {
+        version: 67,
+      },
+      data: {},
+    };
+
+    const newStorage = await migration68.migrate(oldStorage);
+    expect(newStorage.meta).toStrictEqual({
+      version: 68,
+    });
+  });
+
+  it('should migrate all data', async () => {
+    const oldStorage = {
+      meta: {
+        version: 67,
+      },
+      data: getOldState(),
+    };
+
+    const newStorage = await migration68.migrate(oldStorage);
+    expect(newStorage).toMatchObject({
+      meta: {
+        version: 68,
+      },
+      data: {
+        FooController: { a: 'b' },
+        PermissionController: { subjects: expect.any(Object) },
+        PermissionLogController: {
+          permissionActivityLog: expect.any(Object),
+          permissionHistory: expect.any(Object),
+        },
+        SubjectMetadataController: { subjectMetadata: expect.any(Object) },
+      },
+    });
+    expect(newStorage.PermissionsController).toBeUndefined();
+    expect(newStorage.PermissionsMetadata).toBeUndefined();
+  });
+
+  it('should migrate the PermissionsController state', async () => {
+    const oldStorage = {
+      meta: {},
+      data: {
+        PermissionsController: getOldState().PermissionsController,
+      },
+    };
+
+    const newStorage = await migration68.migrate(oldStorage);
+    const { PermissionController } = newStorage.data;
+
+    expect(PermissionController).toStrictEqual(
+      getExpectedPermissionControllerState(),
```

Comment:
This test is very difficult to read. The size of the fixtures plus the fact that they're both declared elsewhere is leaving me flipping back and forth between them to see what this is asserting.

Reply:
Yeah, fair enough. I inlined the expected state getters here: f8a006ec8

Combined with pruned fixture from cd4d3a5, is that good enough?